packages/components/data-table/src/use-manual-column-resizing-reducer.js

```diff
@@ -83,9 +83,13 @@ const useManualColumnResizing = (tableRef) => {
   // if the table element has been rendered and we haven't yet measured the columns
   if (state.tableRef.current && !state.sizes) {
     const renderedColumnMeasurements = [];
-    state.tableRef.current.querySelectorAll('th').forEach((header, index) => {
-      renderedColumnMeasurements[index] = header.getBoundingClientRect().width;
+    state.tableRef.current.querySelectorAll('th').forEach((header) => {
+      renderedColumnMeasurements.push({
+        key: header.getAttribute('data-testid').split(/-(.+)/)[1],
```

Comment:
I'm wondering if we should add the id as a different data-attribute (`data-id` or `data-item-id`) since the `data-testid` is specifically for testing purposes and I'm not sure if we should mix things, even if it's kinda convenient here.

Reply:
I also thought of having `id (data-id)` on the element level, and I think it makes sense. It would also relieve us of having this `str.split()` logic and the `testid`s can be changed in the future and won't affect the rest of the implementation